Merge changes from beta 20.7.0.1, hotfix 20.6.1, editorialized release notes, and beta 20.7.0.2

mokagio opened the pull request:
### 20.7.0.1

Notice that a new beta wasn't necessary because, so far, we only had changes to the release notes and the release automation.

The reason I'm shipping one, though, is that the Jetpack `20.7.0.0` binary was rejected during TestFlight review with:

> ITMS-90433: Invalid Swift Support - The file libswift_Concurrency.dylib doesn't have the correct code signature. Make sure you're using the correct signature, rebuild your app using the current public (GM) version of Xcode, and resubmit it. Don't just modify the code signature of libswift_Concurrency.dylib.

I think/hope this was a random TestFlight issue, but the only way to verify is to submit a new binary, with a new version, for review. I don't want to leave this verification till later in the release cycle in case the issue is legitimately on our end.

### 20.7.0.2

This PR was still open when new changes landed in the release branch after `20.7.0.1`, so I queue `20.7.0.2` to it.

Includes:

- [`gutenberg-mobile` release 1.82.1 which includes a List v2 fix](https://github.com/wordpress-mobile/WordPress-iOS/pull/19290) by @SiobhyB 
- [A fix for the Jetpack review URL](https://github.com/wordpress-mobile/WordPress-iOS/pull/19299) by @guarani 

---

- [x] Updates to the localization files (`.strings` and Fastlane metadata) automatically pulled for the new translations that have already been approved in GlotPress
- [x] Version update in `.xcconfig`
- [x] Diffs from the PRs that made it into this beta

mokagio committed:
Address a RuboCop violation: Remove a redundant `()` occurrence

mokagio committed:
Rename `branch_to_build` parameter into `branch`

This makes the lane parameter using the same name as the
`buildkite_trigger_build` action, avoiding confusion in the rare cases
we need to call it manually.

If you called the `trigger_release_build` or `trigger_beta_build` lane
without a parameter, you'll get prompted to pass a `branch` one, by the
underlying `buildkite_trigger_build` action.

But if you next time called the lane with a `branch` parameter, you'd
still get prompted to pass one, because the lane expected
`branch_to_build` , but this was not made clear by the messages Fastlane
prompted.

Rather than adding extra code for this, I thought aligning the
parameters name would sort out the issue. This leans a bit into the
convention over configuration realm, though. There's nothing stopping
future developers to reintroduce the same confusion by renaming the
parameter either at the lane or action level. Given the limited userbase
of this code and the relative slow pace at which it changes, I think
that's an acceptable tradeoff to have convenience for the time being.

mokagio committed:
Add default `branch` value in lanes to trigger release builds

See conversation at
https://github.com/wordpress-mobile/WordPress-iOS/pull/19281/files#r962698176

mokagio committed:
Leverage default parameter in `trigger_..._build` usage

This makes the code a tiny bit clearer.

mokagio committed:
Centralize logic to compute release branch name

Commit:
Update Gutenberg ref

Commit:
Update Gutenberg ref

mokagio committed:
Bump version number

mokagio committed:
Merge remote-tracking branch 'origin/release/20.7' into mokagio/align-parameters-names-in-buildkite-actions

Commit:
Update Gutenberg ref

Commit:
Merge pull request #19286 from wordpress-mobile/gutenberg/integrate_release_1.81.2

Integrate gutenberg-mobile release 1.81.2

mokagio committed:
Rename `branch_to_build` parameter into `branch` (#19281)

mokagio committed:
Merge `origin/release/20.7` into `merge/release-20.6.1-into-20.7`

Got conflicts in the `Podfile` and `Podfile.lock` and the version
`xcconfig` files.

As usual with hotfixes, I resolved the version conflict by keeping the
release version value, 20.7.

For the pod conflict, that was between Gutenberg 1.81.2 and 1.82.0. I
kept 1.82.0 under the assumption the fix in 1.81.2 will land in the
1.82.x release series soon.

SiobhyB committed:
Release script: Update gutenberg-mobile ref

SiobhyB committed:
Update Gutenberg Mobile reference

SiobhyB committed:
Fix syntax used to reference Gutenberg Mobile

SiobhyB committed:
Update syntax used to reference Gutenberg Mobile

SiobhyB committed:
Apply changes to Podfile

mokagio committed:
Merge hotfix 20.6.1 into release 20.7 (#19292)

mokagio committed:
Manually update `release_notes.txt` with late entry for 20.7

mokagio committed:
Update app translations – `Localizable.strings`

mokagio committed:
Update WordPress metadata translations

mokagio committed:
Update Jetpack metadata translations

mokagio committed:
Bump version number

mokagio committed:
Add editorialized release notes for Jetpack 20.7

mokagio committed:
Add editorialized release notes for WordPress 20.7

mokagio committed:
Update metadata strings

mokagio committed:
Update metadata strings

AliSoftware commented on WordPress/Jetpack/Resources/release_notes.txt:
```diff
@@ -1 +1,3 @@
-* [*] Block Editor: Update link colors in action sheets from green to blue [https://github.com/WordPress/gutenberg/pull/42996]
+Links in action sheets (those pop-up menus on your device that bring up extra options) are now blue instead of green.
```
~~Sorry for missing this earlier and before this was sent for editorial, but looking at https://github.com/WordPress/gutenberg/pull/42996 it seems that this change in Gutenberg might only after iOS after all and this shouldn't have been added to the Android release notes?~~

cc @SiobhyB @dcalhoun ?

Never mind, I though I was reviewing the PR for Android… this is the PR for iOS, so all good 😊

AliSoftware replied:
Though this entry in the release notes for **Jetpack** still feels odd, because it's about Jetpack and yet we're saying we updated colors from green to blue, not the other way around…?!

Or @mokagio did you accidentally swapped the WP and JP release notes when you copied the Editorialized versions in the wrong files?

SiobhyB replied:
For posterity, just x-posting the comment here to confirm all looks good with the notes: https://github.com/wordpress-mobile/WordPress-Android/pull/17145#discussion_r965703697

SiobhyB committed:
Update to latest Gutenberg Mobile reference

SiobhyB committed:
Reference latest Gutenberg Mobile tag

SiobhyB committed:
Merge pull request #19290 from wordpress-mobile/gutenberg/integrate_release_1.82.1

Integrate gutenberg-mobile release 1.82.1

Commit:
Fix Jetpack app review URL

Ratings submitted for the Jetpack app were being submitted to the WordPress app's store listing. This happened because the app was using the incorrect URL to open the App Store app review screen.
While the app correctly defined the URL for the Jetpack app, it was then being overwritten at runtime by a URL fetched from an endpoint. This endpoint was only returning a URL for the WordPress app, so it overwrote the Jetpack app's URL with the WordPress app's one.

This PR mitigates the problem on the Jetpack app by skipping the endpoint request for the Jetpack app only. This doesn't change behavior on the WordPress app.
The impact of this change is that we can no longer remotely disable in-app requests for reviews. However, I don't recall this every being used.
The priority here is to restore the Rate Us functionality for the Jetpack app, so follow-up work can look for a more permanenent fix.

Commit:
Reverted unnecessary change

Commit:
Merge pull request #19299 from wordpress-mobile/fix/rate-us-for-jetpack-app

Fix Jetpack app review URL

mokagio committed:
Update app translations – `Localizable.strings`

mokagio committed:
Update WordPress metadata translations

mokagio committed:
Update Jetpack metadata translations

mokagio committed:
Bump version number

AliSoftware commented on WordPress/Classes/System/WordPressAppDelegate.swift:
```diff
@@ -437,9 +437,11 @@ extension WordPressAppDelegate {
-        utility.checkIfAppReviewPromptsHaveBeenDisabled(success: nil, failure: {
-            DDLogError("Was unable to retrieve data about throttling")
-        })
+        if AppConfiguration.isWordPress {
+            utility.checkIfAppReviewPromptsHaveBeenDisabled(success: nil, failure: {
+                DDLogError("Was unable to retrieve data about throttling")
+            })
+        }
```
Comes from https://github.com/wordpress-mobile/WordPress-iOS/pull/19299/files which was merged in `release/20.7` after code freeze 👍

AliSoftware commented on WordPress/Resources/fr.lproj/Localizable.strings:
```diff
@@ -9920,6 +9922,45 @@ translators: %s: Select control option value e.g: \"Auto, 25%\". */
 
+/* Add an author prompt for the jetpack prologue */
+"jetpack.prologue.prompt.addAuthor" = "Ajouter un auteur";
+
+/* Build a site prompt for the jetpack prologue */
+"jetpack.prologue.prompt.buildSite" = "Construire un site";
+
+/* Check notifications prompt for the jetpack prologue */
+"jetpack.prologue.prompt.checkNotifications" = "Vérifier vos notifications";
+
+/* Share on Facebook prompt for the jetpack prologue */
+"jetpack.prologue.prompt.fbShare" = "Partager sur Facebook";
+
+/* Fix a security issue prompt for the jetpack prologue */
+"jetpack.prologue.prompt.fixSecurity" = "Résoudre un incident de sécurité";
+
+/* Post a photo prompt for the jetpack prologue */
+"jetpack.prologue.prompt.postPhoto" = "Publier une photo";
+
+/* Read an article prompt for the jetpack prologue */
+"jetpack.prologue.prompt.readArticle" = "Lire un article";
+
+/* Respond to comments prompt for the jetpack prologue */
+"jetpack.prologue.prompt.respondComments" = "Répondre aux commentaires";
+
+/* Restore a backup prompt for the jetpack prologue */
+"jetpack.prologue.prompt.restoreBackup" = "Rétablir une sauvegarde";
+
+/* Search for plugins prompt for the jetpack prologue */
+"jetpack.prologue.prompt.searchPlugins" = "Rechercher des extensions";
+
+/* Search for posts or sites prompt for the jetpack prologue */
+"jetpack.prologue.prompt.searchPosts" = "Rechercher des articles ou sites";
+
+/* Watch your stats prompt for the jetpack prologue */
+"jetpack.prologue.prompt.watchStats" = "Consulter vos statistiques";
+
+/* Write a blog prompt for the jetpack prologue */
+"jetpack.prologue.prompt.writeBlog" = "Tenir un blog";
+
```
Keys using reverse-DNS notation 💪💪💪 
<img alt="Feels Good Meme" src="https://user-images.githubusercontent.com/216089/189337645-a8259117-a4b1-4dbf-8db6-fb7c6c34bf20.png" width="150px" />

AliSoftware commented on fastlane/jetpack_metadata/ru/name.txt:
```diff
@@ -0,0 +1 @@
+Jetpack — конструктор сайтов
```
New translation finally fitting length limits? :nice: 👍

AliSoftware reviewed (approved):
Checked the diff and saw this also includes:
 - [x] Tooling PR https://github.com/wordpress-mobile/WordPress-iOS/pull/19281
 - [x] Beta fix https://github.com/wordpress-mobile/WordPress-iOS/pull/19299
 - [x] Gutenberg `1.82.1`, following recent hotfix
 - [x] Editorialized Release Notes in `AppStoreStrings.po`
 - [x] Some new `.strings` already translated (including new App Store metadata for `ru`)

:shipit:
